pidtree_bcc/plugin.py

```diff
@@ -0,0 +1,48 @@
+import importlib
+
+class BasePlugin:
+    def __init__(self, args):
+        self.validate_args(args)
+
+    def process(self, event):
+        """ Process the `event` dict, add in additional metadata and
+        return a dict """
+        raise NotImplementedError("Required method `process` has not been implemented by {}" % self.__name__)
+
+    def validate_args(self, args):
+        """ Not required, override in inheriting class if you want to
+        use this """
+        pass
+
+
+class Plugins:
+    def __init__(self, plugin_dict, plugin_dir="pidtree_bcc.plugins"):
+        """ `plugin_dict` is a dict where the keys are plugin names and
+        the value for each key is another dict of kwargs. Each key must
+        match a `.py` file in the plugin directory and the kwargs can
+        be validated on initialization of the plugin """
+        self._plugins = []
+        for plugin_name, plugin_args in plugin_dict.items():
+            if not plugin_args.get("enabled", False):
+                next
```

Comment:
s/next/continue

Reply:
Forgive me for sounding dumb but what's the difference?

Reply:
`next` is [a builtin function](https://docs.python.org/3/library/functions.html#next) to read iterators, it just does nothing in loops.

Reply:
Fixed